front-end-tools/src/components/InitComponent.tsx

```diff
@@ -0,0 +1,643 @@
+
+                                            const schema = btoa(
+                                                new Uint8Array(arrayBuffer).reduce((data, byte) => {
+                                                    return data + String.fromCharCode(byte);
+                                                }, '')
+                                            );
+                                            setUploadedModuleSchemaBase64(schema);
+                                        } else {
+                                            setUploadError2('Upload schema file is undefined');
+                                        }
+                                    }}
+                                />
+                                <Form.Text />
+                            </Form.Group>
+                        )}
+                        {!useModuleReferenceFromStep1 && uploadedModuleSchemaBase64 && (
+                            <div className="actionResultBox">
+                                Schema in base64:
+                                <div>{uploadedModuleSchemaBase64.toString().slice(0, 30)} ...</div>
+                            </div>
+                        )}
+                        {uploadError2 !== undefined && <Alert variant="danger"> Error: {uploadError2}. </Alert>}
+                        {schemaError !== undefined && <Alert variant="danger"> Error: {schemaError}. </Alert>}
+                        {inputParameterTemplate && (
+                            <>
+                                <br />
+                                <br />
+                                <div className="actionResultBox">
+                                    Parameter Template:
+                                    <pre>{JSON.stringify(JSON.parse(inputParameterTemplate), undefined, 2)}</pre>
+                                </div>
+                            </>
+                        )}
+                        <br />
+                        <Form.Group className="mb-3">
+                            <Form.Label>Select input parameter type:</Form.Label>
+                            <Select
+                                options={INPUT_PARAMETER_TYPES_OPTIONS}
+                                {...form.register('inputParameterType')}
+                                onChange={(e) => {
+                                    form.setValue('inputParameterType', e?.value);
+                                    form.setValue('inputParameter', undefined);
+
+                                    setParsingError(undefined);
+                                }}
+                            />
+                            <Form.Text />
+                        </Form.Group>
+
+                        {(inputParameterType === 'number' || inputParameterType === 'string') && (
+                            <Form.Group className="mb-3">
+                                <Form.Label> Add your input parameter ({inputParameterType}):</Form.Label>
```

Comment:
It's a bit odd that this is here when the error is as well.

![image](https://github.com/Concordium/concordium-smart-contract-tools/assets/105212/46f2ad2b-c85d-4f10-bb2c-ec6f6304a14e)

Whatever I put into that input field will be unusable. So it'd be better to not have it.

Reply:
I agree with you that as long as we do not have a valid schema, the front end will not be able to parse the input parameter and create a valid transaction (with that input parameter). 

On the other hand, it is not good to wipe out the input parameter field too often (e.g. hiding it every time an error occurs). 
Imagine you have a complex object inputted as your input parameter there and when typing your 'entrypoint' name you create a typo, suddenly an error occurs and even though you could easily just correct the typo in the `entrypoint` name field, you have to start again to input your input parameter object in this field.

On the other hand, it allows us to display the example objects/array.